### DISCUSSION TITLE
Should _DEBUG always get defined in configurations with debug symbols?

### DISCUSSION THREAD

**Discussion by rewrking** [chalet-org/chalet#266] [Q&A]
Requesting some feedback from anyone who has an opinion on this:

Right now, `_DEBUG` gets defined in all `*-pc-windows-msvc` builds when a build configuration has debugging symbols. This is not consistent with GNU-based toolchains where it's not defined automatically for you, and something I've been on the fence about changing.

Should this get defined by other toolchains as well? I have a tendency to add `"defines[:debug]": [ "_DEBUG" ]` in projects anyway, and use that to define my own project-specific debug macro (or `NDEBUG` if using assert) in the PCH somewhere. 

Additional notes:
- `_MT`, `_DLL` and `_DEBUG`  could all get defined by MSVC via `/MTd` and `/MDd`. If compiling windows targets via clang, they get set manually right now to mimic the behavior
- `__OPTIMIZE__` is the way to query this on GCC/clang

Curious about other's thoughts on this topic.

👍 Yes, automatically define `_DEBUG`
👎 No, undefine _DEBUG in `pc-windows-msvc`
👀 No, leave this behaviors as-is

**Comment by paulross80** [comment]
I've always defined my own `macro`/`const static bool` on every project, just to make sure it will work on every platform
